presto-hive/src/test/java/com/facebook/presto/hive/TestStreamingAggregationPlan.java

```diff
@@ -0,0 +1,185 @@
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(ORDERS, LINE_ITEM, CUSTOMER, NATION),
+                ImmutableMap.of("experimental.pushdown-subfields-enabled", "true"),
+                Optional.empty());
+    }
+
+    @Test
+    public void testUnsortedTable()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE ds_customer WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  partitioned_by=array['ds'], \n" +
+                    "  format = 'DWRF' ) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM customer LIMIT 1000\n");
+
+            // even with streaming aggregation enabled, non-ordered table that can't be applied streaming aggregation would use hash based aggregation
+            assertPlan(streamingAggregationEnabled(),
+                    "SELECT custkey, COUNT(*) FROM ds_customer \n" +
+                            "WHERE ds = '2021-07-11' GROUP BY 1",
+                    node(OutputNode.class, node(ExchangeNode.class, node(AggregationNode.class, node(ExchangeNode.class, node(ProjectNode.class, node(AggregationNode.class, node(TableScanNode.class))))))));
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS ds_customer");
+        }
+    }
+
+    @Test
+    public void testBucketedAndSortedBySameKey()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE ds_customer WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['custkey'], partitioned_by=array['ds'], \n" +
+                    "  format = 'DWRF' ) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM customer LIMIT 1000\n");
+
+            queryRunner.execute("INSERT INTO ds_customer \n" +
+                    "SELECT *, '2021-07-12' as ds FROM tpch.sf1.customer LIMIT 1000");
+
+            // default: hash based aggregation
+            assertPlan("SELECT custkey, COUNT(*) FROM ds_customer \n" +
+                            "WHERE ds = '2021-07-11' GROUP BY 1",
+                    node(OutputNode.class, node(ExchangeNode.class, node(AggregationNode.class, node(ExchangeNode.class, node(ProjectNode.class, node(AggregationNode.class, node(TableScanNode.class))))))));
+
+            // streaming aggregation enabled
+            assertPlan(streamingAggregationEnabled(),
+                    "SELECT custkey, COUNT(*) FROM ds_customer \n" +
+                            "WHERE ds = '2021-07-11' GROUP BY 1",
+                    node(OutputNode.class, node(ExchangeNode.class, node(AggregationNode.class, node(TableScanNode.class)))));
```

Comment:
Is there a way to verify the preGroupedColumns field of the aggregation node to make sure streaming is enabled?